network/src/transport.rs

```diff
@@ -301,21 +319,33 @@ where
         chain_id: ChainId,
         application_protocols: SupportedProtocols,
     ) -> Self {
-        let mut own_handshake = HandshakeMsg::new(chain_id, network_context.network_id().clone());
-        own_handshake.add(SUPPORTED_MESSAGING_PROTOCOL, application_protocols);
+        // build supported protocols
+        let mut supported_protocols = BTreeMap::new();
+        supported_protocols.insert(SUPPORTED_MESSAGING_PROTOCOL, application_protocols);
+
+        // get network public key
         let identity_pubkey = identity_key.public_key();
 
+        // auth mode
+        // TODO(mimoo): should we build this based on the networkid, and not on trusted peers
         let auth_mode = match trusted_peers.as_ref() {
             Some(trusted_peers) => HandshakeAuthMode::mutual(trusted_peers.clone()),
             None => HandshakeAuthMode::ServerOnly,
         };
 
+        // create upgrade context
```

Comment:
please avoid these type of comments they're the first thing people delete later

Reply:
but why? I actually really appreciate it when people leave these comments, it makes the code much more readable.